internal/slicer/pathtrack.go

```diff
@@ -0,0 +1,212 @@
+package slicer
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"io/fs"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/canonical/chisel/internal/db"
+	"github.com/canonical/chisel/internal/deb"
+	"github.com/canonical/chisel/internal/fsutil"
+	"github.com/canonical/chisel/internal/lib/sortedset"
+	"github.com/canonical/chisel/internal/strdist"
+)
+
+type pathTracker interface {
+	// initialization
+	addSlicePath(slice, path string)
+	addSliceGlob(slice, glob string)
+	// extraction
+	onData(source string, size int64) (deb.ConsumeData, error)
+	onCreate(source, target, link string, mode fs.FileMode) error
+	// non-extracted content
+	addTarget(target, link string, mode fs.FileMode, data []byte)
+	// mutation scripts
+	markMutated(target string)
+	// until paths
+	removeTarget(target string)
+	// reconcile changes above
+	updateTargets(root string) error
+	// write results
+	updateDB(addToDB AddToDB) error
+}
```

Comment:
I understand the purpose of the comments here is grouping the methods in this interface. Does it add enough value to understanding the interface that breaks the Go convention of commenting interface methods? See <https://tip.golang.org/doc/comment#func>